db/compaction/compaction_iterator.cc

```diff
@@ -124,6 +124,7 @@ CompactionIterator::CompactionIterator(
 CompactionIterator::~CompactionIterator() {
   // input_ Iteartor lifetime is longer than pinned_iters_mgr_ lifetime
   input_->SetPinnedItersMgr(nullptr);
+  status_.PermitUncheckedError();
```

Comment:
Can you explain why you added ```PermitUncheckedError``` instead of checking the status through ```status()``` in the test?

Reply:
If user need the status, they will call `const Status& status()` to get the status during iterator calling. So user might call it and might not. Also, check the status_ in `status()` returns the same status to the user. So I may prefer to add `PermitUncheckedError` to the destructor.

Reply:
Thank you for clarifying the doubt.